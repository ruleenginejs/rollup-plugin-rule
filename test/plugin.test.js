const { rollup } = require('rollup');
const path = require('path');
const rule = require('../');

const fixturesDir = path.resolve(__dirname, 'fixtures');

describe('rollup-plugin-rule', () => {
  it('should output compiled rule in es module', async () => {
    const output = await build('basic.js');
    expect(output).toMatchSnapshot();
  });

  it('should throw error when bad rule schema', async () => {
    expect.assertions(1);
    let message = null;
    try {
      await build('bad-schema.js');
    } catch (e) {
      message = e.message;
    }
    expect(message).toMatchSnapshot();
  });
});

async function build(file) {
  const bundle = await rollup({
    input: path.join(fixturesDir, file),
    external: ["@ruleenginejs/runtime"],
    plugins: [
      rule()
    ]
  });
  const { output } = await bundle.generate({ format: 'cjs' });
  return output[0].code;
}
