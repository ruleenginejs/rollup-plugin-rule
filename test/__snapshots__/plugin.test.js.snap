// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rollup-plugin-rule should output compiled rule in es module 1`] = `
"'use strict';

var runtime = require('@ruleenginejs/runtime');

var handler_3 = (context, next) => {
  next();
};

const pipeline = new runtime.Pipeline();
const startStep_1 = new runtime.StartStep({
  id: 2,
  name: \\"Start\\",
  ports: {
    in: [\\"default\\"],
    out: [\\"default\\"]
  }
});
const singleStep_2 = new runtime.SingleStep({
  id: 5,
  name: \\"New Step\\",
  handler: handler_3,
  ports: {
    in: [\\"default\\"],
    out: [\\"default\\"]
  }
});
const endStep_4 = new runtime.EndStep({
  id: 8,
  name: \\"End\\",
  ports: {
    in: [\\"default\\"],
    out: [\\"default\\"]
  }
});
startStep_1.connectTo(singleStep_2, \\"default\\", \\"default\\");
singleStep_2.connectTo(endStep_4, \\"default\\", \\"default\\");
pipeline.add(startStep_1, singleStep_2, endStep_4);

pipeline.execute();
"
`;

exports[`rollup-plugin-rule should throw error when bad rule schema 1`] = `"Schema invalid: [{\\"instancePath\\":\\"\\",\\"schemaPath\\":\\"#/required\\",\\"keyword\\":\\"required\\",\\"params\\":{\\"missingProperty\\":\\"steps\\"},\\"message\\":\\"must have required property 'steps'\\"}]"`;
